salt/client/__init__.py

```diff
@@ -567,10 +567,14 @@ def cmd_batch(
         if 'gather_job_timeout' in kwargs:
             opts['gather_job_timeout'] = kwargs['gather_job_timeout']
 
+        eauth=kwargs.get('eauth', {})
```

Comment:
Very close! `kwargs` here may contain `token`, `username`, `password`, and `eauth`. That last one is the name of the backend (e.g., `pam`) rather than the dictionary of those values that the `Batch` class is expecting.